### PR TITLE
Network switcher component

### PR DESCRIPTION
**Pull request by kenchangh**
- Switches between the Ethereum mainnet to Avalanche mainnet
- Degrades gracefully to an unconnected wallet state when the user selects an unsupported network (ie fetching vault data will return the empty data)
- Redirects the user back to the home page when user lands on vault that does not exist on the chain

### CODE REVIEW

File: shared/src/utils/chainSwitching.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { CHAINID } from "../constants/constants";
+
+export const switchChains = async (
+  provider: providers.Web3Provider,
+  chainId: number
+) => {
+  const hexChainId = "0x" + chainId.toString(16);
+
+  try {
+    await provider.provider.request!({
```
Comment (fqlx) [inline shared/src/utils/chainSwitching.ts:12]: Can you remove the non null assertion?

File: webapp/src/components/Wallet/AccountStatus.tsx [hunk 2]
```diff
@@ -500,8 +510,13 @@ const AccountStatus: React.FC<AccountStatusProps> = ({
...
               renderCopiedButton()
             )}
-            {chainId && renderMenuItem(`OPEN IN ${BLOCKCHAIN_EXPLORER_NAME[chainId]}`, handleOpenEtherscan)}
+            {chainId &&
+              renderMenuItem(
+                `OPEN IN ${BLOCKCHAIN_EXPLORER_NAME[chainId]}`,
+                handleOpenEtherscan
+              )}
             {renderMenuItem("DISCONNECT", handleDisconnect)}
+            {renderAirdropButton()}
```
Comment (fqlx) [inline webapp/src/components/Wallet/AccountStatus.tsx:519]: FYI, These really should be a component and declared like `<AirdropBotton />` Inline rendering components like this causes a performance hit because the virtual DOM can't follow the reconciliation process and screws up the render tree.

https://stackoverflow.com/questions/67454966/whats-the-difference-between-functions-that-render-jsx-vs-declaring-components

File: webapp/src/components/Wallet/AccountStatus.tsx [hunk 3]
```diff
@@ -420,6 +421,15 @@ const AccountStatus: React.FC<AccountStatusProps> = ({
     return <WalletCopyIcon className="far fa-clone" state={copyState} />;
   };
 
+  const renderAirdropButton = () => {
```
Comment (fqlx) [inline webapp/src/components/Wallet/AccountStatus.tsx:424]: What does this have to do with the network switcher..?

File: shared/src/utils/chainSwitching.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      method: "wallet_switchEthereumChain",
+      params: [{ chainId: hexChainId }],
+    });
+  } catch (switchError: any) {
+    // This error code indicates that the chain has not been added to MetaMask.
+    if (switchError.code === 4902) {
```
Comment (fqlx) [inline shared/src/utils/chainSwitching.ts:18]: Ugh are these codes typed somewhere?  Probably not?

File: shared/src/constants/chainParameters.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  chainId: string; // A 0x-prefixed hexadecimal string
+  chainName: string;
+  nativeCurrency: {
+    name: string;
+    symbol: string; // 2-6 characters long
+    decimals: 18;
+  };
+  rpcUrls: string[];
+  blockExplorerUrls?: string[];
+  iconUrls?: string[]; // Currently ignored.
```
Comment (fqlx) [inline shared/src/constants/chainParameters.ts:11]: Why add it if we aren't using it?

File: shared/src/constants/constants.ts [hunk 6]
```diff
@@ -19,19 +19,41 @@ export enum CHAINID {
...
   [CHAINID.ETH_MAINNET]: "mainnet",
   [CHAINID.ETH_KOVAN]: "kovan",
   [CHAINID.AVAX_FUJI]: "fuji",
   [CHAINID.AVAX_MAINNET]: "avax",
 };
 
+// TODO: Remove the isProduction check when enabling avalanche
+export const ENABLED_CHAINID: CHAINID[] = isProduction()
+  ? [CHAINID.ETH_MAINNET]
+  : [CHAINID.ETH_MAINNET, CHAINID.AVAX_MAINNET];
```
Comment (fqlx) [inline shared/src/constants/constants.ts:33]: nit: Could add `isDevelopment()` so it works in localhost but not a big deal.

File: webapp/src/components/Wallet/AccountStatus.tsx (hunk 3 above)
Comment (kenchangh) [reply webapp/src/components/Wallet/AccountStatus.tsx]: This was part of the mocks because we removed the airdrop button from the header

File: shared/src/constants/chainParameters.ts (hunk 5 above)
Comment (kenchangh) [reply shared/src/constants/chainParameters.ts:11]: Just copy pasting the entire type into the definition

File: shared/src/utils/chainSwitching.ts (hunk 4 above)
Comment (kenchangh) [reply shared/src/utils/chainSwitching.ts:18]: Added some comments to clarify this

File: shared/src/constants/constants.ts (hunk 6 above)
Comment (kenchangh) [reply shared/src/constants/constants.ts:33]: This is how we normally do feature flagging

**Review by fqlx — APPROVED** [review]

**Review by cjinghong — APPROVED** [review]
